### ISSUE TITLE
`frank` could support reverse ranking

### ISSUE THREAD

**Issue by hope-data-science** [Rdatatable/data.table#5489]
I try to work with dates using `frank` but find that the minus symbol seems not working:

``` 
library(data.table)
dates <- c("02/27/92", "02/27/92", "01/14/92", "02/28/92", "02/01/92")
as.Date(dates, "%m/%d/%y") -> dates2
frank(dates2,ties.method = "min")
#> [1] 3 3 1 5 2
frank(-dates2,ties.method = "min")
#> Error in `-.Date`(dates2): unary - is not defined for "Date" objects
```

Is there a way to make it work?

Thanks.

**Comment by hope-data-science** [comment]
I understand, and have done so. The issue is you can not rank date reversely using minus symbol, and frank does not provide a method for reverse ranking. Is it a necessity? This problem is open.

**Comment by hope-data-science** [comment]
I know the `frankv` has a `order` parameter, maybe this could be applied to `frank` as well.

**Comment by jangorecki** [comment]
Yes, frank uses frankv, so just passing this new arg seems to be a small change.

**Comment by Nj221102** [comment]
For this tackling this issue my approach is to :-
1.  Modify the ‘frank’ function to include an ‘order’ parameter, similar to ‘frankv’. 
2. Implement the functionality similar to frankv, which will allow users to specify the desired order of ranking, including the option for reverse ranking

**Comment by MichaelChirico** [comment]
I would prefer to do this as `setorder()` and `forder()` do -- `-` is intercepted in `frank()` and passed on to `order=` automatically.

**Comment by Mukulyadav2004** [comment]
Hi @MichaelChirico , may I work on this .
Following the `setorder()` and `forder() `approach, I will modify `frank()` to intercept the `-` symbol, extract variables and their intended orders, and then pass them to `frankv()`.
